day_two_guide/topics/managing_docker_registries.adoc

```diff
@@ -0,0 +1,674 @@
+== Procedure
+
+. Modify `/etc/containers/registries.conf` configuration file to add the
+localhost insecure registry:
++
+----
+registries:
+  - registry.access.redhat.com
+  - my.registry.example.com
+  - docker.io
+insecure_registries:
+  - localhost:5000
+block_registries:
+  - all
+----
++
+Prior to 3.6, modify the `/etc/sysconfig/docker` configuration file to add the
+localhost:
++
+----
+ADD_REGISTRY="--add-registry=registry.access.redhat.com --add-registry=my.registry.example.com --add-registry=docker.io --add-registry=localhost:5000"
+INSECURE_REGISTRY="--insecure-registry=localhost:5000"
+BLOCK_REGISTRY='--block-registry=all'
+----
+
+. Restart the `docker` daemon to use the registry:
++
+----
+$ sudo systemctl restart docker.service
+----
++
+Restarting the `docker` daemon causes the `docker` containers to be restarted.
+
+. Run a Docker registry pod at `localhost`:
++
+----
+$ sudo docker run -p 5000:5000 registry:2
+----
+
+. Pull an image:
++
+----
+$ sudo docker pull openshift/hello-openshift
+----
+
+. Tag the image:
++
+----
+$ sudo docker tag docker.io/openshift/hello-openshift:latest localhost:5000/hello-openshift-local:latest
+----
+
+. Push the image to the local registry:
++
+----
+$ sudo docker push localhost:5000/hello-openshift-local:latest
+----
+
+. Using the Ansible installer, this can be configured using the
+`openshift_docker_additional_registries`, `openshift_docker_blocked_registries`
+and `openshift_docker_insecure_registries` variables in the `Ansible` hosts
+file:
++
+----
+openshift_docker_additional_registries=registry.access.redhat.com,my.registry.example.com,localhost:5000
+openshift_docker_insecure_registries=localhost:5000
+openshift_docker_blocked_registries=all
+----
+
+==== Authenticated registries
+
+Using authenticated registries with `docker` requires the `docker` daemon to log
+in to the registry. With {product-title}, a different set of steps must be
+performed, because the users cannot run `docker login` commands on the host.
+Authenticated registries can be used to limit the images users can pull or who
+can access the external registries.
+
+If an external `docker` registry to requires authentication, the
+process to authentication is to create a special secret in the
+project that uses that registry and then use that secret to perform
+the `docker` operations.
+
+[discrete]
+== Procedure
+
+. Create a `dockercfg` secret in the project where the user is going to log in
+to the `docker` registry:
++
+----
+$ oc project <my_project>
+$ oc secrets new-dockercfg <my_registry> --docker-server=<my.registry.example.com> --docker-username=<username> --docker-password=<my_password> --docker-email=<me@example.com>
+----
+
+. If a `.dockercfg` file exists, create the secret using the `oc` command:
++
+----
+$ oc secrets new <my_registry> .dockercfg=<.dockercfg>
+----
+
+. Populate the `$HOME/.docker/config.json` file:
++
+----
+$ oc secrets new <my_registry> .dockerconfigjson=<.docker/config.json>
```

Comment:
do we need add --type=kubernetes.io/dockercfg here?

Reply:
I've just tried without the `--type` and it seems to work:
```
[root@master-0 ~]# docker login https://index.docker.io/v1/
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: eminguez
Password: 
Login Succeeded
[root@master-0 ~]# cat .docker/config.json 
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "<REDACTED>"
		}
	}
}[root@master-0 ~]# oc secrets new index.docker.io .dockerconfigjson=.docker/config.json 
secret/index.docker.io
[root@master-0 ~]# docker images
REPOSITORY                                                  TAG                 IMAGE ID            CREATED             SIZE
docker.io/busybox                                           latest              5b0d59026729        9 days ago          1.146 MB
registry.access.redhat.com/openshift3/logging-fluentd       v3.7                82ef3baf533b        7 weeks ago         561.8 MB
registry.access.redhat.com/openshift3/ose-service-catalog   v3.7                fa19a1c6a700        7 weeks ago         268.8 MB
registry.access.redhat.com/openshift3/ose-deployer          latest              db1ada96417a        7 weeks ago         1.059 GB
registry.access.redhat.com/openshift3/ose-pod               v3.7.14             22467c8e19ed        7 weeks ago         208.9 MB
[root@master-0 ~]# docker tag docker.io/busybox docker.io/eminguez/deleteme:latest
[root@master-0 ~]# docker push docker.io/eminguez/deleteme:latest
The push refers to a repository [docker.io/eminguez/deleteme]
4febd3792a1f: Mounted from library/busybox 
latest: digest: sha256:4cee1979ba0bf7db9fc5d28fb7b798ca69ae95a47c5fecf46327720df4ff352d size: 527
```